src/tools/SimpleTraversability.hpp

```diff
@@ -0,0 +1,192 @@
+#define __MAPS_SIMPLE_TRAVERSABILITY_HPP_
+
+#include <maps/grid/GridMap.hpp>
+#include <maps/grid/TraversabilityGrid.hpp>
+
+namespace maps { namespace tools
+{
+    /** @brief Configuration parameters for the SimpleTraversability operator
+     *
+     * See SimpleTraversability for more information on the use of these
+     * parameters.
+     */
+    struct SimpleTraversabilityConfig
+    {
+        /** The maximum slope the robot is able to traverse */
+        double maximumSlope;
+
+        /** Number of classes in the output map */
+        int classCount;
+
+        /**
+        * Required ground clearance, in meters. A cell is classified as an
+        * obstacle if the maximum step around that cell is higher than the
+        * ground clearance
+        */
+        double groundClearance;
+
+        /**
+        * Traversable passages that are narrower (in m) than this will be
+        * closed
+        */
+        double minPassageWidth;
+
+        /**
+        * Clearance (in m) around obstacles. When set to a value larger than
+        * 0, all obstacles will be grown by a border of given width.
+        */
+        double obstacleClearance;
+
+        SimpleTraversabilityConfig()
+            : maximumSlope(0)
+            , classCount(0)
+            , groundClearance(0)
+            , minPassageWidth(0)
+            , obstacleClearance(0)
+        {
+        }
+
+        SimpleTraversabilityConfig(
+                double maximumSlope,
+                int classCount,
+                double groundClearance,
+                double minPassageWidth = 0,
+                double obstacleClearance = 0)
+
+            : maximumSlope(maximumSlope)
+            , classCount(classCount)
+            , groundClearance(groundClearance)
+            , minPassageWidth(minPassageWidth)
+            , obstacleClearance(obstacleClearance)
+        {
+        }
+
+    };
+
+    /**
+     * @brief Classification of terrain into symbolic traversability classes, based on
+     * geometric constraints
+     *
+     * For now, the modalities that are used are:
+     *
+     * <ul>
+     * <li>maximum slope
+     * <li>maximum step size
+     * </ul>
+     *
+     * The (very simple) model maps the local slope from [@a maximumSlope, 1] to [0, 1].
+     * The resulting value is then quantified in @a classCount intervals.
+     * Probability will be set to 1 for every classification beside CLASS_UNKNOWN.
+     *
+     * It outputs a TraversabilityGrid in which each cell has an integer value,
+     * this integer value being the traversability class for the cell. Since
+     * CLASS_UNKNOWN (=0) and CLASS_OBSTACLE (=1) are reserved, the resulting
+     * traversability class is in [2, 2 + classCount - 1]
+     *
+     * If one of the modalities is missing, it is simply ignored
+     */
+    class SimpleTraversability {
+
+        enum INPUT_DATA {
+            SLOPE,
+            MAX_STEP,
+            INPUT_COUNT
+        };
+
+    public:
```

Comment:
Try to have only one private, protected and public scope per class definition.

Reply:
Ok.